## First we explore the spray.csv

In [3]:
import pandas as pd
import numpy as np
spray = pd.read_csv('../datasets/spray.csv')

In [2]:
spray.head()

Date        Time   Latitude  Longitude
0  2011-08-29  6:56:58 PM  42.391623 -88.089163
1  2011-08-29  6:57:08 PM  42.391348 -88.089163
2  2011-08-29  6:57:18 PM  42.391022 -88.089157
3  2011-08-29  6:57:28 PM  42.390637 -88.089158
4  2011-08-29  6:57:38 PM  42.390410 -88.088858

In [3]:
def explore(df):
    print(df.shape)
    print("")
    print(df.head())
    print("")
    print(df.dtypes)
    print("")
    print(df.isnull().sum())
    print("")
    print(df.describe())
    
explore(spray)

(14835, 4)

         Date        Time   Latitude  Longitude
0  2011-08-29  6:56:58 PM  42.391623 -88.089163
1  2011-08-29  6:57:08 PM  42.391348 -88.089163
2  2011-08-29  6:57:18 PM  42.391022 -88.089157
3  2011-08-29  6:57:28 PM  42.390637 -88.089158
4  2011-08-29  6:57:38 PM  42.390410 -88.088858

Date          object
Time          object
Latitude     float64
Longitude    float64
dtype: object

Date           0
Time         584
Latitude       0
Longitude      0
dtype: int64

           Latitude     Longitude
count  14835.000000  14835.000000
mean      41.904828    -87.736690
std        0.104381      0.067292
min       41.713925    -88.096468
25%       41.785001    -87.794225
50%       41.940075    -87.727853
75%       41.980978    -87.694108
max       42.395983    -87.586727


In [4]:
#checking the list of date that spraying occured
spray['Date'] = pd.to_datetime(spray['Date'])
spray['Time'] = pd.to_timedelta(spray['Time'])
spray.groupby(['Date']).count()

Time  Latitude  Longitude
Date                                 
2011-08-29    95        95         95
2011-09-07  1530      2114       2114
2013-07-17  2202      2202       2202
2013-07-25  1607      1607       1607
2013-08-08  1195      1195       1195
2013-08-15  2668      2668       2668
2013-08-16   141       141        141
2013-08-22  1587      1587       1587
2013-08-29  2302      2302       2302
2013-09-05   924       924        924

In [5]:
#checking which are the dates that time is null
nans = lambda spray: spray[spray.isnull().any(axis=1)]
nans(spray)

Date Time   Latitude  Longitude
1030 2011-09-07  NaT  41.987092 -87.794286
1031 2011-09-07  NaT  41.987620 -87.794382
1032 2011-09-07  NaT  41.988004 -87.794574
1033 2011-09-07  NaT  41.988292 -87.795486
1034 2011-09-07  NaT  41.988100 -87.796014
1035 2011-09-07  NaT  41.986372 -87.794862
1036 2011-09-07  NaT  41.986228 -87.795582
1037 2011-09-07  NaT  41.984836 -87.793998
1038 2011-09-07  NaT  41.984836 -87.794670
1039 2011-09-07  NaT  41.984884 -87.795198
1040 2011-09-07  NaT  41.982820 -87.793182
1041 2011-09-07  NaT  41.982772 -87.793950
1042 2011-09-07  NaT  41.982772 -87.795006
1043 2011-09-07  NaT  41.983444 -87.795246
1044 2011-09-07  NaT  41.984068 -87.795438
1045 2011-09-07  NaT  41.984596 -87.795534
1046 2011-09-07  NaT  41.984980 -87.795630
1047 2011-09-07  NaT  41.986708 -87.796110
1048 2011-09-07  NaT  41.987524 -87.796302
1049 2011-09-07  NaT  41.988820 -87.796638
1050 2011-09-07  NaT  41.989444 -87.796830
1051 2011-09-07  NaT  41.990068 -87.796974
1052 2011-09-07  NaT  41.988244 -87.797022
1053 2011-09-07  NaT  41.988196 -87.797742
1054 2011-09-07  NaT  41.988196 -87.798366
1055 2011-09-07  NaT  41.988196 -87.798942
1056 2011-09-07  NaT  41.988196 -87.799422
1057 2011-09-07  NaT  41.986372 -87.796494
1058 2011-09-07  NaT  41.986372 -87.797310
1059 2011-09-07  NaT  41.986372 -87.798030
...         ...  ...        ...        ...
1584 2011-09-07  NaT  41.992372 -87.816367
1585 2011-09-07  NaT  41.992468 -87.815839
1586 2011-09-07  NaT  41.992516 -87.815311
1587 2011-09-07  NaT  41.992564 -87.814735
1588 2011-09-07  NaT  41.992612 -87.814303
1589 2011-09-07  NaT  41.992612 -87.813775
1590 2011-09-07  NaT  41.992612 -87.813343
1591 2011-09-07  NaT  41.992660 -87.812767
1592 2011-09-07  NaT  41.992660 -87.812191
1593 2011-09-07  NaT  41.992612 -87.811663
1594 2011-09-07  NaT  41.992660 -87.811279
1595 2011-09-07  NaT  41.992660 -87.810847
1596 2011-09-07  NaT  41.992612 -87.810415
1597 2011-09-07  NaT  41.992660 -87.809455
1598 2011-09-07  NaT  41.992660 -87.809983
1599 2011-09-07  NaT  41.992708 -87.808927
1600 2011-09-07  NaT  41.992708 -87.808399
1601 2011-09-07  NaT  41.992756 -87.807775
1602 2011-09-07  NaT  41.992756 -87.807199
1603 2011-09-07  NaT  41.996164 -87.813727
1604 2011-09-07  NaT  41.995732 -87.813679
1605 2011-09-07  NaT  41.994916 -87.813727
1606 2011-09-07  NaT  41.992948 -87.813775
1607 2011-09-07  NaT  41.996644 -87.811567
1608 2011-09-07  NaT  41.996260 -87.811567
1609 2011-09-07  NaT  41.995876 -87.811615
1610 2011-09-07  NaT  41.995972 -87.810271
1611 2011-09-07  NaT  41.995684 -87.810319
1612 2011-09-07  NaT  41.994724 -87.810415
1613 2011-09-07  NaT  41.993092 -87.810415

[584 rows x 4 columns]

## Spray conclusion:
    spray.shape = (14835, 4)
    584 null values in Time column only, all null values are in 2011-09-07
    However, not all 2011-09-07 time are null values (584/1530 are nulls)
    
    changed Date and Time to datetime and timedelta respectively
    
    2 sprays in 2011, once in Aug, once in Sept 
    8 sprays in 2013, twice in July, 5 times in Aug, once in Sept
    
sidenote: might be linked to weather conditions <br>
(ie hotter in 2013 than 2011, or increased mosquitoes activites hence an increase in sprays)

In [6]:
spray_2 = spray

In [7]:
spray_2.to_csv("../datasets/spray_2.csv", index=False) 


## Now, we move on to explore "weather.csv"

In [8]:
weather = pd.read_csv('../datasets/weather.csv')

In [9]:
weather.head()

Station        Date  Tmax  Tmin Tavg Depart  DewPoint WetBulb Heat Cool  \
0        1  2007-05-01    83    50   67     14        51      56    0    2   
1        2  2007-05-01    84    52   68      M        51      57    0    3   
2        1  2007-05-02    59    42   51     -3        42      47   14    0   
3        2  2007-05-02    60    43   52      M        42      47   13    0   
4        1  2007-05-03    66    46   56      2        40      48    9    0   

   ... CodeSum Depth Water1 SnowFall PrecipTotal StnPressure SeaLevel  \
0  ...             0      M      0.0        0.00       29.10    29.82   
1  ...             M      M        M        0.00       29.18    29.82   
2  ...      BR     0      M      0.0        0.00       29.38    30.09   
3  ...   BR HZ     M      M        M        0.00       29.44    30.08   
4  ...             0      M      0.0        0.00       29.39    30.12   

  ResultSpeed ResultDir  AvgSpeed  
0         1.7        27       9.2  
1         2.7        25       9.6  
2        13.0         4      13.4  
3        13.3         2      13.4  
4        11.7         7      11.9  

[5 rows x 22 columns]

In [10]:
explore(weather)

(2944, 22)

   Station        Date  Tmax  Tmin Tavg Depart  DewPoint WetBulb Heat Cool  \
0        1  2007-05-01    83    50   67     14        51      56    0    2   
1        2  2007-05-01    84    52   68      M        51      57    0    3   
2        1  2007-05-02    59    42   51     -3        42      47   14    0   
3        2  2007-05-02    60    43   52      M        42      47   13    0   
4        1  2007-05-03    66    46   56      2        40      48    9    0   

   ... CodeSum Depth Water1 SnowFall PrecipTotal StnPressure SeaLevel  \
0  ...             0      M      0.0        0.00       29.10    29.82   
1  ...             M      M        M        0.00       29.18    29.82   
2  ...      BR     0      M      0.0        0.00       29.38    30.09   
3  ...   BR HZ     M      M        M        0.00       29.44    30.08   
4  ...             0      M      0.0        0.00       29.39    30.12   

  ResultSpeed ResultDir  AvgSpeed  
0         1.7        27       9.2  
1       

In [11]:
#expand the Date column to Year,Month,Day to check on the period of data and see if there are any missing days/months
from pandas import Timestamp

weather['Date'] = pd.to_datetime(weather['Date'])
weather['Year'] = weather['Date'].dt.year
weather['Month'] = weather['Date'].dt.month
weather['Day'] = weather['Date'].dt.day

# Show three rows
weather.head(3)

Station       Date  Tmax  Tmin Tavg Depart  DewPoint WetBulb Heat Cool  \
0        1 2007-05-01    83    50   67     14        51      56    0    2   
1        2 2007-05-01    84    52   68      M        51      57    0    3   
2        1 2007-05-02    59    42   51     -3        42      47   14    0   

   ... SnowFall PrecipTotal StnPressure SeaLevel ResultSpeed ResultDir  \
0  ...      0.0        0.00       29.10    29.82         1.7        27   
1  ...        M        0.00       29.18    29.82         2.7        25   
2  ...      0.0        0.00       29.38    30.09        13.0         4   

  AvgSpeed  Year Month  Day  
0      9.2  2007     5    1  
1      9.6  2007     5    1  
2     13.4  2007     5    2  

[3 rows x 25 columns]

In [12]:
#Creating dummy variables for the weather CodeSum data
weather['CodeSum'] = weather['CodeSum'].apply(str.split)
for i, list_ in enumerate(weather['CodeSum']):
    for item in list_:
        if item not in weather.columns:
            weather[item] = 0
            weather.set_value(col=item,index=i,value=1)
        else:
            weather.set_value(col=item,index=i,value=1)


/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:7: FutureWarning: set_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead
  import sys
/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:9: FutureWarning: set_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead
  if __name__ == '__main__':


In [13]:
#replacing codes with other values
weather = weather.replace('M',0)
weather = weather.replace('-',np.nan)
weather = weather.replace(' ',np.nan)
weather = weather.replace('T',0.0005)

In [14]:
#dropping these columns since they are used for winter purposes,
#hence their values are zero and empty for the other weather station
weather = weather.drop(['Depth','Water1','SnowFall'],axis=1)

In [15]:
#To fill missing Tavg data with Average Temp.
for i in range(weather.shape[0]):
    if weather.loc[i,'Tavg'] == 0:
        weather.loc[i,'Tavg'] = int(np.round((weather.loc[i,'Tmax'] + weather.loc[i,'Tmin'])/2))
    else:
        weather.loc[i,'Tavg'] = weather.loc[i,'Tavg']

In [16]:
columns = ['Tavg',
           'Depart']

for i in columns:
    weather[i] = weather[i].astype(str).astype(float)

### Wet Bulb definition and approach
#source: [https://www.theweatherprediction.com/habyhints/170/] <br>
A quick technique that many forecasters use to determine the wet-bulb temperature <br>
is called the "1/3 rule". The technique is to first find the dewpoint depression (temperature minus dewpoint). <br>
Then take this number and divide by 3. Subtract this number from the temperature.<br>

Since there are only 4 missing wetbulb cells, we will use this estimate<br>
wet bulb might be linked to humidity, we might be able to create a feature engineer a new column<br>

In [17]:
#To estimate the value for 'WetBulb'
for i in range(weather.shape[0]):
    if weather.loc[i,'WetBulb'] == 0:
        weather.loc[i,'WetBulb'] = np.round(weather.loc[i,'Tavg'] - (weather.loc[i,'Tavg'] - weather.loc[i,'Depart'])/3)
                                                                    
    else:
        weather.loc[i,'WetBulb'] = weather.loc[i,'WetBulb']

### Heat/Cool definition and approach
#Heat
A heating degree day (HDD) is a measurement designed to quantify the demand for energy 
needed to heat a building. It is the number of degrees that a day's average temperature 
is below 65o Fahrenheit (18o Celsius), which is the temperature below which buildings 
need to be heated. 

#Cool
A cooling degree day (CDD) is a measurement designed to quantify the demand for energy 
needed to cool buildings. It is the number of degrees that a day's average temperature 
is above 65o Fahrenheit (18o Celsius).

In [18]:
#To estimate the value for 'Heat' and 'Cool'
for i in range(weather.shape[0]):
    if weather.loc[i,'Heat'] == 0:
        if 65 - weather.loc[i,'Tavg'] > 0:
            weather.loc[i,'Heat'] = 65 - weather.loc[i,'Tavg']
        else:
            weather.loc[i,'Heat'] = 0 
    else:
        weather.loc[i,'Heat'] = weather.loc[i,'Heat']

        

for i in range(weather.shape[0]):
    if weather.loc[i,'Cool'] == 0:
        if  weather.loc[i,'Tavg'] - 65 > 0:
            weather.loc[i,'Cool'] = weather.loc[i,'Tavg'] - 65
        else:
            weather.loc[i,'Cool'] = 0 
    else:
        weather.loc[i,'Cool'] = weather.loc[i,'Cool']

In [19]:
#changing traces of precipation total to 0.005 so that we can use the values to analyse the model
weather['PrecipTotal'] = weather['PrecipTotal'].replace(np.nan,'0.00')
weather['PrecipTotal'] = weather['PrecipTotal'].replace('  T','0.005')

In [20]:
#change station pressure np.nan to mean
weather['StnPressure'] = weather['StnPressure'].replace(np.nan,0)
weather['StnPressure'] = weather['StnPressure'].astype(str).astype(float)
weather['StnPressure'] = weather['StnPressure'].replace(0,weather['StnPressure'].mean())

In [21]:
#change sealevel np.nan to mean
weather['SeaLevel'] = weather['SeaLevel'].replace(np.nan,0)
weather['SeaLevel'] = weather['SeaLevel'].astype(str).astype(float)
weather['SeaLevel'] = weather['SeaLevel'].replace(0,weather['StnPressure'].mean())

In [22]:
columns = ['WetBulb',
           'Heat',
           'Cool',
           'PrecipTotal',
           'AvgSpeed',
           'Sunrise',
           'Sunset']

for i in columns:
    weather[i] = weather[i].astype(str).astype(float)

In [23]:
explore(weather)

(2944, 38)

   Station       Date  Tmax  Tmin  Tavg  Depart  DewPoint  WetBulb  Heat  \
0        1 2007-05-01    83    50  67.0    14.0        51     56.0   0.0   
1        2 2007-05-01    84    52  68.0     0.0        51     57.0   0.0   
2        1 2007-05-02    59    42  51.0    -3.0        42     47.0  14.0   
3        2 2007-05-02    60    43  52.0     0.0        42     47.0  13.0   
4        1 2007-05-03    66    46  56.0     2.0        40     48.0   9.0   

   Cool  ...  TS  DZ BCFG  FG+  MIFG  FG  SQ  SN  VCFG  GR  
0   2.0  ...   0   0    0    0     0   0   0   0     0   0  
1   3.0  ...   0   0    0    0     0   0   0   0     0   0  
2   0.0  ...   0   0    0    0     0   0   0   0     0   0  
3   0.0  ...   0   0    0    0     0   0   0   0     0   0  
4   0.0  ...   0   0    0    0     0   0   0   0     0   0  

[5 rows x 38 columns]

Station                 int64
Date           datetime64[ns]
Tmax                    int64
Tmin                    int64
Tavg                 

In [24]:
weather.shape

(2944, 38)

In [25]:
weather_2 = weather.groupby('Date').agg({'Year':'max',
                                         'Month':'max',
                                         'Day':'max',
                                         'Tmax':'mean', 
                                         'Tmin':'mean', 
                                         'Tavg':'mean', 
                                         'Depart': 'sum', 
                                         'DewPoint':'mean',
                                         'WetBulb':'mean',
                                         'Heat':'mean',
                                         'Cool':'mean',
                                         'Sunrise':'mean',
                                         'Sunset':'mean',
                                         'PrecipTotal':'mean',
                                         'StnPressure':'mean',
                                         'SeaLevel':'mean',
                                         'ResultSpeed':'mean',
                                         'ResultDir':'mean',
                                         'AvgSpeed':'mean',
                                         'BR':'max',
                                         'HZ':'max',
                                         'RA':'max',
                                         'TSRA':'max', 
                                         'VCTS':'max', 
                                         'FU':'max',
                                         'TS':'max',
                                         'DZ':'max',
                                         'BCFG':'max',
                                         'FG+':'max',
                                         'MIFG':'max',
                                         'SQ':'max',
                                         'SN':'max',
                                         'VCFG':'max',
                                         'GR':'max'})

In [26]:
explore(weather_2)

(1472, 34)

            Year  Month  Day  Tmax  Tmin  Tavg  Depart  DewPoint  WetBulb  \
Date                                                                        
2007-05-01  2007      5    1  83.5  51.0  67.5    14.0      51.0     56.5   
2007-05-02  2007      5    2  59.5  42.5  51.5    -3.0      42.0     47.0   
2007-05-03  2007      5    3  66.5  47.0  57.0     2.0      40.0     49.0   
2007-05-04  2007      5    4  72.0  50.0  61.0     4.0      41.5     50.0   
2007-05-05  2007      5    5  66.0  53.5  60.0     5.0      38.5     49.5   

            Heat  ...  FU  TS  DZ  BCFG  FG+  MIFG  SQ  SN  VCFG  GR  
Date              ...                                                 
2007-05-01   0.0  ...   0   0   0     0    0     0   0   0     0   0  
2007-05-02  13.5  ...   0   0   0     0    0     0   0   0     0   0  
2007-05-03   8.0  ...   0   0   0     0    0     0   0   0     0   0  
2007-05-04   4.0  ...   0   0   0     0    0     0   0   0     0   0  
2007-05-05   5.0  ... 

In [27]:
columns = ['Sunrise',
           'Sunset']

for i in columns:
    weather_2[i] = weather_2[i].astype(float).astype(int)
    weather_2[i] = weather_2[i].astype(int).astype(str)

In [28]:
#changing Sunrise and Sunset to type - time
weather_2['Sunrise'] = str(0) + weather_2['Sunrise']
weather_2['Sunrise'] = weather_2['Sunrise'].str[0:2] + ":" + weather_2['Sunrise'].str[2:4] + ":00"

weather_2['Sunset'] = weather_2['Sunset'].str[0:2] + ":" + weather_2['Sunset'].str[2:4] + ":00"


weather_2['Sunrise'] = pd.to_timedelta(weather_2['Sunrise'])
weather_2['Sunset'] = pd.to_timedelta(weather_2['Sunset'])

In [29]:
weather_2 = weather_2.reset_index()

In [30]:
weather_2.to_csv("../datasets/weather_2.csv", index=False) 

## Conclusion Weather:
weather.shape = (1472, 36) <br>

change the CodeSum to dummies

# Next, to explore train and test sets

Will merge both data sets, do data cleaning on both.
Then seperate them indivdually again

In [4]:
train = pd.read_csv('../datasets/train.csv')
test = pd.read_csv('../datasets/test.csv')

In [5]:
train.columns

Index(['Date', 'Address', 'Species', 'Block', 'Street', 'Trap',
       'AddressNumberAndStreet', 'Latitude', 'Longitude', 'AddressAccuracy',
       'NumMosquitos', 'WnvPresent'],
      dtype='object')

In [6]:
test.columns

Index(['Id', 'Date', 'Address', 'Species', 'Block', 'Street', 'Trap',
       'AddressNumberAndStreet', 'Latitude', 'Longitude', 'AddressAccuracy'],
      dtype='object')

In [7]:
print(train.shape)
print(test.shape)

(10506, 12)
(116293, 11)


In [8]:
test = test.drop(['Id'], axis=1)

In [9]:
combined = pd.merge(train,test,how='outer')

In [10]:
explore(combined)

NameError: name 'explore' is not defined

In [11]:
#changing Date to type - date
combined['Date'] = pd.to_datetime(combined['Date'])

In [12]:
#taking out the address out of the data frame and store it in address in case we need it
Address = pd.DataFrame([combined.Address, combined.Block, combined.Street, combined.AddressNumberAndStreet, combined.AddressAccuracy])
Address = Address.T

#dropping duplicates
Address = Address.drop_duplicates()
print(Address.shape)

(151, 5)


In [13]:
combined = combined.drop(['Address','Block','Street','AddressNumberAndStreet','AddressAccuracy'],axis=1)

In [14]:
combined.head()

Date                 Species  Trap   Latitude  Longitude  \
0 2007-05-29  CULEX PIPIENS/RESTUANS  T002  41.954690 -87.800991   
1 2007-05-29          CULEX RESTUANS  T002  41.954690 -87.800991   
2 2007-05-29          CULEX RESTUANS  T007  41.994991 -87.769279   
3 2007-05-29  CULEX PIPIENS/RESTUANS  T015  41.974089 -87.824812   
4 2007-05-29          CULEX RESTUANS  T015  41.974089 -87.824812   

   NumMosquitos  WnvPresent  
0           1.0         0.0  
1           1.0         0.0  
2           1.0         0.0  
3           1.0         0.0  
4           4.0         0.0

In [15]:
#change Species into dummy variables
column = ['Species']

combined = pd.get_dummies(data = combined, prefix=column, columns = column, drop_first=True)

In [16]:
combined.head(2)

Date  Trap  Latitude  Longitude  NumMosquitos  WnvPresent  \
0 2007-05-29  T002  41.95469 -87.800991           1.0         0.0   
1 2007-05-29  T002  41.95469 -87.800991           1.0         0.0   

   Species_CULEX PIPIENS  Species_CULEX PIPIENS/RESTUANS  \
0                      0                               1   
1                      0                               0   

   Species_CULEX RESTUANS  Species_CULEX SALINARIUS  Species_CULEX TARSALIS  \
0                       0                         0                       0   
1                       1                         0                       0   

   Species_CULEX TERRITANS  Species_UNSPECIFIED CULEX  
0                        0                          0  
1                        0                          0

In [17]:
#do a groupby to combine the days where there are multiple trap rows
#ie those traps that has more than 50 mosquitos

combined_2 = combined.groupby(['Date','Trap']).agg({'Latitude':'max',
                                                    'Longitude':'max',
                                                    'Species_CULEX PIPIENS':'max',
                                                    'Species_CULEX PIPIENS/RESTUANS':'max',
                                                    'Species_CULEX RESTUANS':'max',
                                                    'Species_CULEX SALINARIUS':'max',
                                                    'Species_CULEX TARSALIS':'max',
                                                    'Species_CULEX TERRITANS':'max',
                                                    'Species_UNSPECIFIED CULEX':'max',
                                                    'NumMosquitos':'sum',
                                                    'WnvPresent':'max'})

In [18]:
combined_2.reset_index(inplace=True)

In [19]:
combined_2.shape

(18771, 13)

In [47]:
#Once we have combined all traps, we can use dummy variables on the Trap column
column = ['Trap']
combined_2 = pd.get_dummies(data = combined_2, prefix=column, columns = column, drop_first=True)

In [48]:
combined_2.head()

Date   Latitude  Longitude  Species_CULEX PIPIENS  \
0 2007-05-29  41.954690 -87.800991                      0   
1 2007-05-29  41.994991 -87.769279                      0   
2 2007-05-29  41.974089 -87.824812                      0   
3 2007-05-29  41.921600 -87.666455                      0   
4 2007-05-29  41.891118 -87.654491                      0   

   Species_CULEX PIPIENS/RESTUANS  Species_CULEX RESTUANS  \
0                               1                       1   
1                               0                       1   
2                               1                       1   
3                               0                       1   
4                               0                       1   

   Species_CULEX SALINARIUS  Species_CULEX TARSALIS  Species_CULEX TERRITANS  \
0                         0                       0                        0   
1                         0                       0                        0   
2                         0                       0                        0   
3                         0                       0                        0   
4                         0                       0                        0   

   Species_UNSPECIFIED CULEX  ...  Trap_T231  Trap_T232  Trap_T233  Trap_T234  \
0                          0  ...          0          0          0          0   
1                          0  ...          0          0          0          0   
2                          0  ...          0          0          0          0   
3                          0  ...          0          0          0          0   
4                          0  ...          0          0          0          0   

   Trap_T235  Trap_T236  Trap_T237  Trap_T238  Trap_T900  Trap_T903  
0          0          0          0          0          0          0  
1          0          0          0          0          0          0  
2          0          0          0          0          0          0  
3          0          0          0          0          0          0  
4          0          0          0          0          0          0  

[5 rows x 160 columns]

In [49]:
combined_2.shape

(18771, 160)

In [50]:
combined_2.to_csv("../datasets/combined_2.csv", index=False) 

## Now to merge with weather

In [51]:
full = pd.merge(combined_2,weather_2, on='Date')

In [52]:
pd.options.display.max_columns = None
full.head()

Date   Latitude  Longitude  Species_CULEX PIPIENS  \
0 2007-05-29  41.954690 -87.800991                      0   
1 2007-05-29  41.994991 -87.769279                      0   
2 2007-05-29  41.974089 -87.824812                      0   
3 2007-05-29  41.921600 -87.666455                      0   
4 2007-05-29  41.891118 -87.654491                      0   

   Species_CULEX PIPIENS/RESTUANS  Species_CULEX RESTUANS  \
0                               1                       1   
1                               0                       1   
2                               1                       1   
3                               0                       1   
4                               0                       1   

   Species_CULEX SALINARIUS  Species_CULEX TARSALIS  Species_CULEX TERRITANS  \
0                         0                       0                        0   
1                         0                       0                        0   
2                         0                       0                        0   
3                         0                       0                        0   
4                         0                       0                        0   

   Species_UNSPECIFIED CULEX  NumMosquitos  WnvPresent  Trap_T002  Trap_T002A  \
0                          0           2.0         0.0          1           0   
1                          0           1.0         0.0          0           0   
2                          0           5.0         0.0          0           0   
3                          0           2.0         0.0          0           0   
4                          0           1.0         0.0          0           0   

   Trap_T002B  Trap_T003  Trap_T004  Trap_T005  Trap_T006  Trap_T007  \
0           0          0          0          0          0          0   
1           0          0          0          0          0          1   
2           0          0          0          0          0          0   
3           0          0          0          0          0          0   
4           0          0          0          0          0          0   

   Trap_T008  Trap_T009  Trap_T011  Trap_T012  Trap_T013  Trap_T014  \
0          0          0          0          0          0          0   
1          0          0          0          0          0          0   
2          0          0          0          0          0          0   
3          0          0          0          0          0          0   
4          0          0          0          0          0          0   

   Trap_T015  Trap_T016  Trap_T017  Trap_T018  Trap_T019  Trap_T025  \
0          0          0          0          0          0          0   
1          0          0          0          0          0          0   
2          1          0          0          0          0          0   
3          0          0          0          0          0          0   
4          0          0          0          0          0          0   

   Trap_T027  Trap_T028  Trap_T030  Trap_T031  Trap_T033  Trap_T034  \
0          0          0          0          0          0          0   
1          0          0          0          0          0          0   
2          0          0          0          0          0          0   
3          0          0          0          0          0          0   
4          0          0          0          0          0          0   

   Trap_T035  Trap_T036  Trap_T037  Trap_T039  Trap_T040  Trap_T043  \
0          0          0          0          0          0          0   
1          0          0          0          0          0          0   
2          0          0          0          0          0          0   
3          0          0          0          0          0          0   
4          0          0          0          0          0          0   

   Trap_T044  Trap_T045  Trap_T046  Trap_T047  Trap_T048  Trap_T049  \
0          0          0          0          0          0          0   
1          0          0

In [53]:
full.shape

(18771, 194)

In [54]:
full.to_csv("../datasets/full.csv", index=False) 

### train/test conclusion: 
train.shape = (10506, 12) <br>
test.shape = (116293, 11) <br>
 
shorten to <br>
combined.shape = (18771, 161) <br>

combined with weather <br>
full.shape = (18771,194)<br>


Condensed the total Mosquitoes collected in each trap <br>
Change Species and Trap to dummies <br>
Stored Addresses in another df <br>
After dropping duplicates, only 151 unique addresses left <br>
Removed all addresses except Latitude and Longitude <br>
Changed Date to date format

In [55]:
train_2 = full.loc[full['Year'].isin([2007,2009,2011,2013])]
test_2 = full.loc[full['Year'].isin([2008,2010,2012,2014])]

In [56]:
train_2 = train_2.reset_index()
test_2 = test_2.reset_index()

In [57]:
train_2 = train_2.drop(['index','Year'], axis = 1)
test_2 = test_2.drop(['index','Year'], axis = 1)

In [58]:
train_2.to_csv("../datasets/train_2.csv", index=False) 
test_2.to_csv("../datasets/test_2.csv", index=False) 

## Lastly, to check the format of the submission

In [59]:
sample = pd.read_csv('../datasets/samplesubmission.csv')

In [60]:
sample.head()

Id  WnvPresent
0   1           0
1   2           0
2   3           0
3   4           0
4   5           0